src/main/java/net/backupcup/mcde/util/EnchantmentUtils.java

```diff
@@ -74,7 +74,7 @@ public static Formatting formatEnchantment(Identifier id) {
         return MCDEnchantments.getConfig().isEnchantmentPowerful(id) ? Formatting.RED : Formatting.LIGHT_PURPLE;
     }
 
-    public static EnchantmentSlots generateEnchantments(ItemStack itemStack, ServerPlayerEntity player) {
+    public static EnchantmentSlots generateEnchantments(ItemStack itemStack, Optional<ServerPlayerEntity> player) {
```

Comment:
Consider the following renaming:
`Optional<ServerPlayerEntity> player` to `Optional<ServerPlayerEntity> optionalServerPlayerEntity`